### PR TITLE
Initial integration with `OboeAPI`

### PR DESCRIPTION
**Pull request by raphael-theriault-swi**
Integrates with `OboeAPI` for sampling and reporting most (but not all) of the metrics. Since the sampling decision function has the same signature as the one from the `Context` API this ends up being a fairly small change. There's also a new OTLP exporter that properly clears the cache.

The initialisation code has been modularised to make it easier to reason about and in general cleaner wrt the flags.

The `experimental` flags have also been renamed to `dev` to clarify their intended use.

### CODE REVIEW

File: packages/sdk/src/sampler.ts [hunk 1]
```diff
@@ -57,10 +58,29 @@ const TRACE_OPTIONS_RESPONSE_TRIGGER_IGNORED = "ignored"
...
+  ) => oboe.DecisionResult
+  private readonly recordMetricsFunction: () => void
+
   constructor(
     private readonly config: SwConfiguration,
     private readonly logger: DiagLogger,
-  ) {}
+    serverlessApi: oboe.OboeAPI | undefined,
+  ) {
+    if (serverlessApi) {
```
Comment (tammy-baylis-swi) [inline packages/sdk/src/sampler.ts:71]: Cool idea!! More compact than what I'm doing at the moment haha

File: packages/sdk/src/metrics/serverless.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  TriggeredTraceCount: lazy(() =>
+    meter.createCounter("trace.service.triggered_trace_count"),
+  ),
+}
+
+export function recordServerlessMetrics(serverlessApi: oboe.OboeAPI) {
+  for (const [name, counter] of Object.entries(counters)) {
+    const method = `consume${name}` as const
+    const value = serverlessApi[method]()
+    if (value !== false) counter.add(value)
```
Comment (tammy-baylis-swi) [inline packages/sdk/src/metrics/serverless.ts:42]: Makes sense with [current comment thread](https://swicloud.atlassian.net/wiki/spaces/NIT/pages/3753116438/AWS+Lambda+Instrumentation+POC?focusedCommentId=3885761076) about sampler rate hopper.

File: packages/sdk/src/sampler.ts [hunk 3]
```diff
@@ -116,6 +136,8 @@ export class SwSampler implements Sampler {
       traceOptions,
       traceState,
     )
+
+    this.recordMetricsFunction()
```
Comment (tammy-baylis-swi) [inline packages/sdk/src/sampler.ts:140]: Also makes sense this is called at sampling decision for a counter. [Asking in doc](https://swicloud.atlassian.net/wiki/spaces/NIT/pages/3753116438/AWS+Lambda+Instrumentation+POC?focusedCommentId=3887530249) for my sanity.

File: packages/sdk/src/otlp-exporter.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { type ExportResult } from "@opentelemetry/core"
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-grpc"
```
Comment (tammy-baylis-swi) [inline packages/sdk/src/otlp-exporter.ts:18]: Has grpc export to the custom otel-collector been working well? Asking for myself as I've only done http export to it so far.

**Review by tammy-baylis-swi — APPROVED** [review]
Lgtm overall!

File: packages/sdk/src/otlp-exporter.ts (hunk 4 above)
Comment (raphael-theriault-swi) [reply packages/sdk/src/otlp-exporter.ts:18]: Mentioned during standup but will also reply here for future reference, yes grpc has been working fine for me

**Pull request merged**